lib/mapping/CMapHeader.cpp

```diff
@@ -127,11 +129,56 @@ CMapHeader::CMapHeader() : version(EMapFormat::VCMI), height(72), width(72),
 	setupEvents();
 	allowedHeroes = VLC->heroh->getDefaultAllowed();
 	players.resize(PlayerColor::PLAYER_LIMIT_I);
+	VLC->generaltexth->addSubContainer(*this);
+}
+
+CMapHeader::~CMapHeader()
+{
+	VLC->generaltexth->removeSubContainer(*this);
 }
 
 ui8 CMapHeader::levels() const
 {
 	return (twoLevel ? 2 : 1);
 }
 
+void CMapHeader::registerMapStrings()
+{
+	auto language = CGeneralTextHandler::getPreferredLanguage();
+	JsonNode data;
+
+	if(translations[language].isNull())
```

Comment:
for now it's impossible since map editor copies all strings to another language when added, but it could be possible optmization and wise to support merge

Reply:
Another problem is that main map language may not be english. There is no way to enforce mapmakers to use some predefined language